### PR TITLE
Replicate medic users meta database

### PR DESCRIPTION
**Pull request by njogz**
Adds the ability to replicate medic-users-meta database to PostgreSQL. 

medic/medic-couch2pg#54

### CODE REVIEW

File: package.json
Comment (njogz) [inline package.json:20]: Using this package to make it easier to test and not have to wait for the accompanying couch2pg change to me merged and released. It will be changed back to `couch2pg` and updated to the correct version number before merging.

File: env.js
Comment (garethbowen) [inline env.js:20]: I think we should remove this environment variable and turn it on for everyone.

File: env.js
Comment (garethbowen) [inline env.js:21]: Just use the `docLimit` from above. No need to have a separate variable.

File: libs/couch2pg.js
Comment (garethbowen) [inline libs/couch2pg.js:28]: I think we should remove the default option here and hardcode 'couchdb' into the calling functions so it's explicit what table is being used.

File: libs/medic-users-meta/migrations/202102191153.do.54-create-couchdb-users-meta-table.sql
Comment (garethbowen) [inline libs/medic-users-meta/migrations/202102191153.do.54-create-couchdb-users-meta-table.sql:30]: NICE!

File: libs/medic-users-meta/migrations/202102191153.do.54-create-couchdb-users-meta-table.sql
Comment (garethbowen) [inline libs/medic-users-meta/migrations/202102191153.do.54-create-couchdb-users-meta-table.sql:43]: Please reach out to Yuvraj to review this schema.

File: tests/replicate_users_meta.js
Comment (garethbowen) [inline tests/replicate_users_meta.js:40]: Ideally we'd also be testing the migration and materialised view. Do you think it would be easy enough to execute that and query the view to ensure it's outputting the rows we expect?

**Review by garethbowen — CHANGES_REQUESTED** [review]
Nice one! A few comments inline but nothing major.

File: env.js
Comment (njogz) [reply env.js:21]: The docs in the users-meta database seem to be quite a bit larger than other docs possibly due to the presence of stack traces. This was causing the app to crash when I tested locally using docLimit so decided to have an additional parameter that can be set lower than the docLimit.

File: env.js
Comment (njogz) [reply env.js:20]: The ticket states that it should be an opt-in feature by default rather than it being on for everyone.

File: tests/replicate_users_meta.js
Comment (njogz) [reply tests/replicate_users_meta.js:55]: Yes. I have added a test for the material views an the migration.

File: env.js
Comment (garethbowen) [reply env.js:21]: Ok! Good call. We probably need to solve this for both cases so they automatically recover when this is set too high but that's outside of this issue.

File: libs/medic-users-meta/migrations/202102191153.do.54-create-couchdb-users-meta-table.sql
Comment (garethbowen) [reply libs/medic-users-meta/migrations/202102191153.do.54-create-couchdb-users-meta-table.sql:43]: Should this be `userview` not `useview`?

Comment (njogz) [reply libs/medic-users-meta/migrations/202102191153.do.54-create-couchdb-users-meta-table.sql:43]: Good catch. I got the code for the views from Yuvraj so I'll confirm with him.

Comment (njogz) [reply libs/medic-users-meta/migrations/202102191153.do.54-create-couchdb-users-meta-table.sql:43]: `useview` is intentional and not a typo. The pattern the team has been following is `Materialized views: useview_` and `Views: formview_`.

File: env.js
Comment (garethbowen) [reply env.js:20]: @michaelkohn and I just discussed this and we've both agreed it should be on for everyone. Sorry for the confusion in the issue!

File: libs/medic-users-meta/migrations/202102191153.do.54-create-couchdb-users-meta-table.sql
Comment (garethbowen) [reply libs/medic-users-meta/migrations/202102191153.do.54-create-couchdb-users-meta-table.sql:43]: Ok!

**Review by garethbowen — CHANGES_REQUESTED** [review]
Nice work.

The one last change is to hardcode it to replicate the user meta data rather than having an environment variable.

File: env.js
Comment (njogz) [reply env.js]: No worries. I have made the change. Please have another look when you can.

**Review by garethbowen — APPROVED** [review]
Great work!

The build failed but it looks unrelated to your work so I've restarted it again.

**Comment by dianabarsan** [comment]
I had to rebase to fix conflicts, as a consequence this comment was orphaned, so linking it here so it isn't lost: https://github.com/medic/medic-couch2pg/commit/4e2e044cbe8d9f52fc19ff5ef9321697548cca56#commitcomment-49150811

File: package.json
Comment (dianabarsan) [reply package.json]: I've published `couch2pg@0.5.0` and reverted this change.

File: env.js
Comment (nomulex) [reply env.js:20]: don't we need a safeNum safety check here?

**Review by nomulex — COMMENTED** [review]
Just curious whether we need to put a check

File: env.js
Comment (garethbowen) [reply env.js:20]: @nomulex Good spotting! As this has already been merged can you please raise a new issue for it?

Comment (nomulex) [reply env.js:20]: Ticket logged [here](https://github.com/medic/cht-couch2pg/issues/99) @garethbowen

**Pull request merged**
